students/duanez2021/lesson05/mailroom_rev03-1.py

```diff
@@ -0,0 +1,224 @@
+#######################################################################################################
+# date          init    desc
+# 20200625      djm     mail room part 1
+# 20200707      djm     mail room part 2 (Dictionary implementations)
+# 20200803      djm     mail room part 3, Lesson 5, exceptions and comprehensions
+# 20200824      djm     mail room part 3 (comments from  natasha-aleksandrova 13 days ago)
+#                       updating part 3 with changes from part 2 (see mailroom_rev02-1.py)
+#               your data structure in lesson04 should be using a dictionary for donors data structure
+# Changes:
+#   donor_db to a dictionary object (see mailroom_rev02-1.py)
+# Replaced:
+#   replaced these functions from from mailroom_rev02-1.py to use dictionary objects
+#   create_required_report
+#   added get_sum_donations
+#   get_number_of_donations
+#   get_avg_donation_amt
+#   check_donor_name
+#   list_donors
+#   add_new_donor
+#   resolve_donor_name
+#   add_donor_amt
+#   prep_individual_email
+#   send_email_to_all
+#
+#   additional changes (for lesson 5)
+#   add_donor_amt (added try-block to catch entry of inappropriate amount values)
+#   prep_individual_email (added try-block to catch error raised by add_donor_amt)
+#   send_email_to_all list (added list comprehension)
+#   main (added try-block to catch error raised by inappropriate menu value entries)
+#
+#   Duane McCollum Python self-paced winter 2020
+#
+#######################################################################################################
+# original donor db, was a list
+
+import sys, os  # imports go at the top of the file
+#print(os.getcwd())
+donor_db = {"William Gates, III": [653772.32, 12.17],
+            "Jeff Bezos": [877.33],
+            "Paul Allen": [663.23, 43.87, 1.32],
+            "Mark Zuckerberg": [1663.23, 4300.87, 10432.0],
+            "Catherine Zeta-Jones": [100.00, 100.00, 1000.00]
+            }
+
+new_prompt = "\n".join(("Choose an action:\n",
+                    "1 - Send a Thank You to a single donor.",
+                    "2 - Create a Report.",
+                    "3 - Send letters to all donors.",
+                    "4 - Quit\n",
+                    ">>> "))
+
+
+thankyou_2_prompt = "Send thank you to (full name)\n(Enter \'list\' for list of donors;\n\'exit\' to return to main menu): "
+thankyou_for_amt_prompt = ("Donation amount: ")
+
+def create_required_report():
+    print('{0: <20}{1} {2: >12} {3} {4: ^10} {5} {6: >13}'.format('Donor Name','|', 'Total Given','|', 'Num Gifts','|','Average Gift'))
+    print('{:''^66}'.format(' '))
+    for idx, donor_name in enumerate(donor_db):
+        t = (donor_name, get_sum_donations(donor_name), get_number_of_donations(donor_name), get_avg_donation_amt(donor_name))
```

Comment:
you're not gaining anything by using a function here - you can directly get the sum: `sum(donor_db[donor_name])`